torch/distributed/_functional_collectives.py

```diff
@@ -145,36 +147,25 @@ def _all_reduce(self, reduceOp, tag, ranks, group_size):
     group = c10d._find_or_create_pg_by_ranks_and_tag(tag, ranks, group_size)
     assert group is not None
 
-    inplace_tensor = self.clone()
+    inplace_tensor = self.clone(memory_format=torch.contiguous_format)
```

Comment:
i _think_ the inductor version is already ok,  since it just creates a new 'FlexibleLayout' for AllReduce and doesn't do anything morally equivalent of '.clone'

Reply:
We ran into this issue when running our e2e model.